core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -4902,6 +4929,18 @@ public interface Config {
 
     /** Sets {@link #convertCorrelateToJoin()}. */
     Config withConvertCorrelateToJoin(boolean convertCorrelateToJoin);
+
+    /** Whether to save the distinct if we know that the input is
+     * already unique; default true. */
+    @Value.Default
+    default boolean redundantDistinct() {
```

Comment:
This flag is a little unintuitive, since it *inhibits* the optimization rather than enabling it.
All the other similar flags are in the opposite way.

Reply:
Good catch.

I changed default value, now we can set `true` to optimize. (Default is `false`)